src/app/clusters/operational-state-server/operational-state-server.cpp

```diff
@@ -0,0 +1,342 @@
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::OperationalState;
+using namespace chip::app::Clusters::OperationalState::Attributes;
+
+using Status = Protocols::InteractionModel::Status;
+
+CHIP_ERROR OperationalStateServer::Init()
+{
+    // Check if the cluster has been selected in zap
+    if (!emberAfContainsServer(mEndpointId, mClusterId))
+    {
+        ChipLogError(Zcl, "Operational State: The cluster with ID %lu was not enabled in zap.", long(mClusterId));
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    ReturnErrorOnFailure(InteractionModelEngine::GetInstance()->RegisterCommandHandler(this));
+
+    VerifyOrReturnError(registerAttributeAccessOverride(this), CHIP_ERROR_INCORRECT_STATE);
+
+    return CHIP_NO_ERROR;
+}
+
+void OperationalStateServer::Shutdown()
+{
+    InteractionModelEngine::GetInstance()->UnregisterCommandHandler(this);
+}
+
+// This function is called by the interaction model engine when a command destined for this instance is received.
+void OperationalStateServer::InvokeCommand(HandlerContext & handlerContext)
+{
+    ChipLogDetail(Zcl, "OperationalState: InvokeCommand");
+    switch (handlerContext.mRequestPath.mCommandId)
+    {
+    case Commands::Pause::Id:
+        ChipLogDetail(Zcl, "OperationalState: Entering handling Pause state");
+
+        HandleCommand<Commands::Pause::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & req) { HandlePauseState(ctx, req); });
+        break;
+
+    case Commands::Resume::Id:
+        ChipLogDetail(Zcl, "OperationalState: Entering handling Resume state");
+
+        HandleCommand<Commands::Resume::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & req) { HandleResumeState(ctx, req); });
+        break;
+
+    case Commands::Start::Id:
+        ChipLogDetail(Zcl, "OperationalState: Entering handling Start state");
+
+        HandleCommand<Commands::Start::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & req) { HandleStartState(ctx, req); });
+        break;
+
+    case Commands::Stop::Id:
+        ChipLogDetail(Zcl, "OperationalState: Entering handling Stop state");
+
+        HandleCommand<Commands::Stop::DecodableType>(handlerContext,
+                                                     [this](HandlerContext & ctx, const auto & req) { HandleStopState(ctx, req); });
+        break;
+    }
+}
+
+void OperationalStateServer::HandlePauseState(HandlerContext & ctx, const Commands::Pause::DecodableType & req)
+{
+    ChipLogDetail(Zcl, "OperationalState: HandlePauseState");
+    Commands::OperationalCommandResponse::Type response;
+    Delegate * delegate = OperationalState::GetOperationalStateDelegate(mEndpointId, mClusterId);
+    GenericOperationalError err(to_underlying(ErrorStateEnum::kNoError));
+    GenericOperationalState opState(to_underlying(OperationalStateEnum::kStopped));
```

Comment:
Seems like we should have a way to construct a GenericOperationalState without providing a state, if we expect to then immediately try to fill it in...